pxr/base/arch/debugger.cpp

```diff
@@ -344,50 +346,58 @@ Arch_DebuggerAttachExecPosix(void* data)
 
 #if defined(ARCH_OS_LINUX)
 
+// Reads /proc/self/status, finds the line starting with "field:", and
+// returns the portion following the ":".
+// Note that the returned string will generally include leading whitespace
 static
-bool
-Arch_DebuggerIsAttachedPosix()
+std::string Arch_ReadProcStatusField(const std::string_view field)
 {
-    // Check for a ptrace based debugger by trying to ptrace.
-    pid_t parent = getpid();
-    pid_t pid = nonLockingFork();
-    if (pid < 0) {
-        // fork failed.  We'll guess there's no debugger.
-        return false;
+    std::ifstream procStatusFile("/proc/self/status");
+    if (!procStatusFile) {
+        ARCH_WARNING("Unable to open /proc/self/status");
+        return std::string();
     }
-
-    // Child process.
-    if (pid == 0) {
-        // Attach to the parent with ptrace() this will fail if the
-        // parent is already being traced.
-        if (ptrace(PTRACE_ATTACH, parent, NULL, NULL) == -1) {
-            // A debugger is probably attached if the error is EPERM.
-            _exit(errno == EPERM ? 1 : 0);
+    for (std::string line; std::getline(procStatusFile, line);) {
+        // the field needs to start with the given fieldLen AND the ':' char
+        if (line.size() < field.size() + 1 ) {
+            continue;
         }
 
-        // Wait for the parent to stop as a result of the attach.
-        int status;
-        while (waitpid(parent, &status, 0) == -1 && errno == EINTR) {
-            // Do nothing
+        if (line.compare(0, field.size(), field) == 0 && line[field.size()] == ':') {
+            // We found our "field:" line
+            return line.substr(field.size() + 2);
```

Comment:
This should be `+ 1` no?  We want the text after the colon?

Reply:
Ah - good catch.  I think the what happened was I initially thought that what followed the `:` was (possibly multiple) space characters - ie, `"Field:    value"`, and decided not to deal with whitespace stripping, added [this comment](https://github.com/PixarAnimationStudios/OpenUSD/pull/3014/files#diff-3bf66466b125291e781fb6bb0328fffe857d6732e71e37103a8e3125f4095ddbR351): 

```
// Note that the returned string will generally include leading whitespace
```

But I then realized what followed was always a tab character - ie, `"Field:\tvalue"`, and I modified the code to just skip the tab character... but forgot to add an explanatory comment (or remove the old comment).

Probably what you've proposed (locale-independent whitespace stripping) is more robust, though.